web/src/components/review-form.jsx

```diff
@@ -120,7 +128,13 @@ export default class ReviewForm extends React.Component {
           </section>
         )) }
 
-        <button type="submit">Finish Review</button>
+        { this.state.pendingSentences && (
+          <p>
+            <strong>Reviews are being uploaded. This can take several minutes depending on the number of sentences added.
+            Please don't close this website.</strong>
+          </p>
+        )}
+        <button type="submit" disabled={this.state.pendingSentences}>Finish Review</button>
```

Comment:
Does this also disable submit by keyboard of the form?

Reply:
while not being 100% sure, I'd say this should be prevented by the browser..